xbmc/filesystem/NFSDirectory.cpp

```diff
@@ -266,25 +264,18 @@ bool CNFSDirectory::GetDirectory(const CURL& url, CFileItemList &items)
 
     iSize = tmpDirent.size;
     bIsDir = tmpDirent.type == NF3DIR;
-    lTimeDate = tmpDirent.mtime.tv_sec;
 
     if (!StringUtils::EqualsNoCase(strName,".") && !StringUtils::EqualsNoCase(strName,"..")
         && !StringUtils::EqualsNoCase(strName,"lost+found"))
     {
-      if(lTimeDate == 0) // if modification date is missing, use create date
-      {
-        lTimeDate = tmpDirent.ctime.tv_sec;
-      }
-
-      long long ll = lTimeDate & 0xffffffff;
-      ll *= 10000000ll;
-      ll += 116444736000000000ll;
-      fileTime.lowDateTime = (DWORD)(ll & 0xffffffff);
-      fileTime.highDateTime = (DWORD)(ll >> 32);
-      KODI::TIME::FileTimeToLocalFileTime(&fileTime, &localTime);
-
       CFileItemPtr pItem(new CFileItem(tmpDirent.name));
-      pItem->m_dateTime=localTime;
+      // thexai pointed that posix and win32 implementations treat this value
+      // as local time to generate consistent directory hashes across platforms
+      // to avoid library re-scans:
+      // https://github.com/xbmc/xbmc/pull/23631
+      pItem->m_dateTime = CDateTime(
+          tmpDirent.mtime.tv_sec != 0 ? tmpDirent.mtime.tv_sec : tmpDirent.ctime.tv_sec)
+          .GetAsLocalDateTime();
```

Comment:
Not think `GetAsLocalDateTime` works on Windows because same reasons of previous problems: DST bug, only works on recent Windows 10 builds, etc.

Maybe use for now if defined(TARGET_WINDOWS) and use `CWIN32Util::fileTimeToLocalFileTime`